File: mcp-server-supabase/check_uploads.py
# -*- coding: utf-8 -*-
import os
import sys
from supabase import create_client, Client
from dotenv import load_dotenv
from datetime import datetime

# Ensure UTF-8 encoding
sys.stdout.reconfigure(encoding='utf-8')

# Load environment variables
load_dotenv()

# Initialize Supabase client
url = os.getenv('SUPABASE_URL')
key = os.getenv('SUPABASE_SERVICE_ROLE_KEY')
supabase = create_client(url, key)

def check_recent_uploads():
    print('Ì∂ºÔ∏è Recent Image Uploads:')
    print('=======================')
    
    try:
        # Fetch recent uploads
        response = supabase.table('part_searches').select(
            'id, image_name, image_url, created_at, user_id'
        ).order('created_at', desc=True).limit(10).execute()
        
        # Check if we have data
        if not response.data:
            print('‚ùó No recent uploads found.')
            return
        
        # Print upload details
        for index, upload in enumerate(response.data, 1):
            print(f'{index}. Image Details:')
            print(f'   Name: {upload.get(image_name, N/A)}')
            print(f'   URL: {upload.get(image_url, N/A)}')
            print(f'   Date: {datetime.fromisoformat(upload[created_at]).strftime(%Y-%m-%d %H:%M:%S)}')
            print(f'   User ID: {upload.get(user_id, N/A)}')
            print('')
    
    except Exception as e:
        print(f'‚ùå Error: {str(e)}')

# Run the function
check_recent_uploads()

File: mcp-server-supabase/check-uploads.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

const supabase = createClient(
  process.env.SUPABASE_URL,
  process.env.SUPABASE_SERVICE_ROLE_KEY
);

async function checkRecentUploads() {
  console.log('Ì∂ºÔ∏è Checking Recent Image Uploads');
  console.log('================================');
  
  try {
    const { data: searches, error } = await supabase
      .from('part_searches')
      .select('id, user_id, image_name, image_url, created_at, confidence_score, predictions')
      .order('created_at', { ascending: false })
      .limit(15);
    
    if (error) {
      console.error('‚ùå Error fetching searches:', error.message);
      return;
    }
    
      console.log('‚ùó No recent uploads found.');
      return;
    }
    
    console.log();
    
    searches.forEach((search, index) => {
      console.log();
      console.log();
      console.log();
      console.log();
      console.log();
      console.log();
      
      if (search.predictions && search.predictions.length > 0) {
        console.log('   Ì¥Æ Predictions:');
        search.predictions.forEach((pred, predIndex) => {
          console.log();
        });
      } else {
        console.log('   Ì¥Æ No predictions found');
      }
      
      console.log(''); // Empty line for readability
    });
    
  } catch (error) {
    console.error('‚ùå Unexpected error:', error.message);
  }
}

checkRecentUploads();
